packages/cli/test/integration.js

```diff
@@ -3675,3 +3675,140 @@ test('[vc link] should support the `--project` flag', async t => {
     formatOutput(output)
   );
 });
+
+test('vercel.json projectSettings should apply to prompts', async t => {
+  const directory = path.join(__dirname, 'fixtures/unit/vercel-json-overrides');
+  const vercelJsonPath = path.join(directory, 'vercel.json');
+  const vercelJsonOriginal = await readFile(vercelJsonPath, 'utf8');
+
+  function assertDeployment(deployment, statusCode) {
+    t.is(
+      deployment.exitCode,
+      statusCode,
+      formatOutput({
+        stderr: deployment.stderr,
+        stdout: deployment.stdout,
+      })
+    );
+  }
+
+  async function deploy() {
+    const deployment = await execa(binaryPath, [
+      directory,
+      ...defaultArgs,
+      '--public',
+    ]);
```

Comment:
We also need to test a single setting such as 

```
{
  projectSettings: {
    outputDirectory: "foo"
  }
}
```

Make sure it overrides one but allows defaults of the others